build_runner/lib/src/asset/file_based.dart

```diff
@@ -50,7 +50,7 @@ class FileBasedAssetReader implements RunnerAssetReader {
     for (var file in files) {
       // TODO(jakemac): Where do these files come from???
       if (path.basename(file.path).startsWith('._')) continue;
-      yield _fileToAssetId(file, packageNode);
+      yield _fileToAssetId(file as File, packageNode);
```

Comment:
Similar to the AssetId.deserialize constructor - the reified type of `files` is `Iterable<FilesystemEntity>` - maybe the downcast is allowed in the `as` though? I think with strong mode runtime semantics we can't do that case.

We what we could do is add the extra `.map((e) => e as File)` though. This is the same kind of thing where the ergonomics of Iterable.where with an `is` check kind of suck.